### ISSUE TITLE
[v4-docs] wrong `content` in docsearch meta

### ISSUE THREAD

**Issue by siriwatknp** [mui/mui-x#2664]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

docsearch team mentioned in [this PR](https://github.com/algolia/docsearch-configs/pull/4635#issuecomment-923720310) that some of the links in v4 has wrong content.

Only the urls for data-grid has wrong content.

```html
<meta name="docsearch:version" content="master">
```

### Expected behavior 🤔

It should be

```html
<meta name="docsearch:version" content="v4">
```

### Steps to reproduce 🕹

Steps:

1. checkout https://v4.mui.com/components/data-grid/
2. in element devtool, search for `docsearch`
3.
4.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Comment by oliviertassinari** [comment]
@siriwatknp I work on this in #2657

**Comment by oliviertassinari** [comment]
I think that the next step for this issue is to update https://github.com/mui-org/material-ui/blob/v4.x/docs/public/_redirects#L151-L159 to point to the `docs-v4` branch.

---

Also note that I fixed a different issue with Algolia on #2655.

**Comment by cherniavskii** [comment]
It's set to "v4" now:
<img width="357" alt="Screenshot 2022-06-29 at 13 19 05" src="https://user-images.githubusercontent.com/13808724/176434502-1c71f722-fe67-4121-a1d7-c55ca8aa0425.png">
